### ISSUE TITLE
Split synth track into two different tracks (synth and effect)

### ISSUE THREAD

**Issue by fpesari** [muse-sequencer/muse#988]
Hello,

could the synth track be split into two different tracks, "Synth" and "Effect"? Finding effects under Synth > LV2 (Effects) is a bit counterintuitive, and also does not make it very clear that Effects can be automated.

Of course, if a MIDI track is created right below an Effect track, it should not set the Effect as its instrument. This is another problem with having effects in a Synth track: right now the effect (as a Synth track) is set as the instrument of the MIDI track below it! The Effect tracks would only be used to route audio through them.

Thank you for reading in any case!

**Comment by kybos** [comment]
> Effects were added as a subset of synths in the arranger to allow for multi input/output effects. Effects in the rack are only stereo capable, while in the arranger we allow more.

That's interesting. I have also been wondering why there are also effects in the synth selections (and why there are also synths in the effect rack selections). All that must be pretty confusing from user perspective. And it has a lot of unwanted side-effects (like the one described in the post above).
On the other hand, I have the impression that the distinction between synth and effect (which is apparently obligatory in the plugin APIs) is not as clear in practice as it may seem. There are audio signal processors that can act both as synth and effect. There are plugin hosts available as plugins (like Carla) that can also host both types, and would probably have a special category/attribute in an ideal word, etc.

**Comment by fpesari** [comment]
> On the other hand, I have the impression that the distinction between synth and effect (which is apparently obligatory in the plugin APIs) is not as clear in practice as it may seem. There are audio signal processors that can act both as synth and effect. There are plugin hosts available as plugins (like Carla) that can also host both types, and would probably have a special category/attribute in an ideal word, etc.

While true, MusE already does this distinction by separating Synths and Effects in the context menu. I think that, at least for LV2, lv2:InstrumentPlugin in the Turtle file should be enough to tell one from the other :smile_cat:

**Comment by terminator356** [comment]
Yes I know it seems awkward. Effects as synth tracks. But it was the easiest way we could give access to all
 the effect's input and output channels since our effects rack is only stereo maximum 2 channels,
 while our synth tracks allow multi-channel input and output.
But also, some effects may actually have midi input or output, which is also something we don't have in
 our audio effects rack, but is readily available as a synth track.

To 'fix' the situation, 1) All our 'audio track' types would need to become true multi-channel and therefore
 the effect rack too, and 2) our effects rack would likely need to have midi inputs and outputs.
At the moment only our synth tracks (which are audio track type) allow true multi-channel audio input and output.
(Channels beyond 2 need to be routed separately in and out by the user.)

Oh, and there are effects that are 'midi effects' (arpeggiators etc.), which complicates things further.
We don't really have a 'midi effects rack' in our midi mixer strips at the moment.

**Comment by spamatica** [comment]
This ticket got derailed, sorry :)

The fix you are outlining Tim, while correct, seems to me is more work than it is worth. We are a stereo application and the odd case where more input/outputs or midi control is needed we can solve by instantiating them like synths. To me that makes sense.
Except for midi-only effects, that is a different beast and we simply do not support it...
 
Better documentation on why we support for effects in the arranger makes sense though. And making the changes suggested by the ticket seem like a good thing.

**Comment by terminator356** [comment]
Sorry I didn't see the original suggestion of making effects tracks.

**Comment by kybos** [comment]
Back to my point I was trying to make above (and either was not clear enough or the point is wrong?):

If we introduce a special effect track type (which then obviously can't be used as synth for a midi track, or what else would be the difference?), what about e.g. the Carla Rack VST, which is declared as effect? I suppose you should be able to use it as a synth track, so here the distinction makes little sense.

And to the second point: If the multi in/out is the only reason for having effects as synths - what is the reason for having synths in the effect rack plugin selection? This is confusing as well.

**Comment by fpesari** [comment]
As a user, I think having a single Synth/Effect kind of track and separate entries for Synths and Effects in the context menu would be great too!

**Comment by kybos** [comment]
1. I understand that effects as tracks is a workaround for a very special case (multi I/O) and as such rarely needed. So it's totally misleading for an average user, who should use the effect rack for this as the first choice.
Which means from UX viewpoint: It should be hidden from the standard workflow. Removed from the menus (see item 3.), and in the synth selection popup only synths should be shown as a default. With the possibility to select effects if really desired of course.

2. I suppose the effect selection popup (in the effect rack) should not show the synths. And be renamed from Plugins to Effects to make it clear? And include a hint as to what to do if you need  a multi i/o effect?

3. As for the synths/effects menu in general: I think it should be removed completely in the mid term (or contain only the user Favorites at least). It can get really huge if you have lots of plugins installed, which means bad UX. The new synth selection dialog was meant to be a more user-friendly replacement.
@fpesari You prefer the menus over the synth selection dialog? Why exactly? Or are you using an older version  of MusE without the dialog?

**Comment by fpesari** [comment]
3. I actually prefer the dialog + menus for favorites only @kybos. I mentioned the menus perhaps because I was used to them, but actually recognize that the dialog is better (I have dozens of plugins and the menus are unusable anyway). I am using the latest version of MusE

**Comment by spamatica** [comment]
Yes, if you put it like that I agree on all three :).

The dialog is superior. I use the menu out of old habit... A Favourites menu would be a good addition though, or do we have that already? Can't check right now :P

And yes, adding Effects from the dialog (by specifically selecting this option) sounds like a good way to allow this special case. And we could give some help text (tool tip maybe) as to why you can add effects there, and why you probably should use the Rack in other cases.

**Comment by terminator356** [comment]
Whatever you decide to do, be careful. We've been down this road several times in the past.
Consider the following cases:

1) Vocoders, tricky to classify. A vocoder will have one set of audio inputs for the signal.
 It may provide another set of audio inputs for the exciter, **or** it might provide midi input to trigger
 a built-in synth for the exciter, **or** it might even allow both or either with a setting.
When audio inputs are chosen as the exciter, the vocoder is perfectly at home in our effects rack,
 despite there being no midi input.

2) Generators. These generate sound by themselves, with or without the aid of midi. Even though they
 might accept midi and may be classified as synths, they still might be at home in our effects rack and work just fine.

3) Effects with midi automation. Some effects accept midi purely for assignable automation, and do not
 actually require midi to function. They would be just fine in our effects rack.
I have an LV2 U-He reverb classified as a synth that accepts midi (for parameter automation I presume).
Yet it should work just fine in our rack without midi.

As for effects as synth tracks, we could eliminate some - but not all.
Proposal: Rename such menu items like "LV2 effects" or "VST effects" as "xxx **multi-channel** effects",
 and only include effects that warrant inclusion.
Examine the effect's number of audio inputs and outputs. If the number of audio inputs is > 2 and/or the
 number of audio outputs > 2, then include it.
Maybe that would make things a bit clearer as to why we include effects as synth tracks.

**Comment by kybos** [comment]
Right, I will better not touch the plugin popup (in the effect rack), other then adding a tooltip or something, that multi i/o effects should be created as pseudo-synth tracks if more than 2 i/o are needed.

> As for effects as synth tracks, we could eliminate some - but not all.
> Proposal: Rename such menu items like "LV2 effects" or "VST effects" as "xxx **multi-channel** effects",
> and only include effects that warrant inclusion.
> Examine the effect's number of audio inputs and outputs. If the number of audio inputs is > 2 and/or the
> number of audio outputs > 2, then include it.

But then, what about effects with 2 audio ins/outs and a midi input?

**Comment by kybos** [comment]
I merged the changes I proposed above (except point 2).
The synth selection dialog now shows only synths as default.
Selecting a synth is now very fast and easy, the top choice is always pre-selected and can be accepted just by pressing Enter. No mouse usage necessary.
There are new root-level submenus for Favorites and Recently used synths.
The "old" nested submenus have been removed (this even improved the start-up time perceptibly in my case, as this complex submenu structures had been set up and cached at start).

**Comment by fpesari** [comment]
I don't think this one deserves to be closed! :smile_cat:
